src/libraries/System.Runtime.InteropServices/tests/System.Runtime.InteropServices.UnitTests/System/Runtime/InteropServices/StandardOleMarshalObjectTests.cs

```diff
@@ -0,0 +1,24 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Xunit;
+
+namespace System.Runtime.InteropServices.Tests
+{
+    [ConditionalClass(typeof(PlatformDetection), nameof(PlatformDetection.IsBuiltInComEnabled))]
+    public class StandardOleMarshalObjectTests
+    {
+        [Fact]
+        public void CanGetIDispatchOfDerivedObject()
+        {
+            IntPtr disp = Marshal.GetIDispatchForObject(new DerivedObject());
```

Comment:
This isn't the same thing as performing a QI as described by the OP. I would prefer to follow the approach that users would most often go. Perform a QI on the CCW's `IUnknown` or `IDispatch` from this API. After the below assert simply call `Marshal.QueryInterface(...)`, validate success and then `Marshal.Release()` again.

Reply:
The OP doesn't talk about performing a QI. It talks about calling this method directly. The repro fails on this call in the original issue.